### DISCUSSION TITLE
Looking for large JSCAD examples.

### DISCUSSION THREAD

**Discussion by briansturgill** [jscad/OpenJSCAD.org#1082] [General]
My original plan was to use my OpenSCAD projects translated to CSCAD for timing tests, but that would require then translating them to JSCAD, which is a fair bit of (otherwise unnecessary) work. Are there large JSCAD examples available?
The examples I see in the source tree are generally trivial.
I would really like to do timing tests on real-world JSCAD projects if they are available.
Also, any chance you have some sort of gargantuan, does everything, torture test available?

Also, I'm torn about translating some parts of JSCAD, in particular, I'm curious about real-world usages of:
scission
align
3D convex hulls
3D version of things in expand (they seem to be missing/partial).
connectors
Anything to do with Path2.

Also, why were radians chosen in calls like rotate?
A CAD package makes real world things. Measuring in radians (such as with a protractor) is nearly impossible
due to the number of digits you need. Additionally few people have good visualization of radians, but many do with degrees.
(My early career was for the Departments of Mathematical Scienes. While Mathematicians do use radians by default, I assure you
that if they are traveling the wrong way down a road, they do a "180", not a "PI".)

And finally, why are Poly3 and Poly2 considered "geometries"?

**Comment by hrgdavor** [comment]
I would say that size of the project is less important. A simple example with few boolean operations on high-count meshes can easily slow down script execution.

**Comment by z3dev** [comment]
Performance tests are at https://github.com/jscad-community/jscad-performance

FYI, I'm going to review these again, and possibly add a few more.

**Reply by briansturgill** [reply]
I'll give those a try next week.

**Comment by z3dev** [comment]
Also, I'm torn about translating some parts of JSCAD, in particular, I'm curious about real-world usages of:
- scission, align ,3D convex hulls, 3D version of things in expand (they seem to be missing/partial), connectors, anything to do with path2

It's up to you but these won't go away in JSCAD. Although you may not use these, there are several users using those functions, even if broken.

Personally, I have written align() several times in the past.

**Reply by briansturgill** [reply]
I asked about align because I'm having trouble making a good C# API for it. But it looks like I'll have to try harder...

**Comment by z3dev** [comment]
> Also, why were radians chosen in calls like rotate?

Consistency basically. Where do you switch from degrees to radians? Also, all the trig functions in Javascript Math use radians. And therefore, JSCAD uses radians.

**Comment by z3dev** [comment]
> And finally, why are Poly3 and Poly2 considered "geometries"?

Arrays (maths), Objects (geometries) is the general policy.

**Comment by gilboonet** [comment]


**Reply by briansturgill** [reply]
Thanks very much for these... My big problem with scission is that I have no idea what it should do. I mean how does it decide where to cut? What happens if you give it multiple arguments?
The documentation is terrible...
https://openjscad.xyz/dokuwiki/doku.php?id=en%3adesign_guide_scission&msclkid=4cca5cf3ce6711ec9b2a4292cbc9ae58

**Reply by z3dev** [reply]
Basically, scission splits complex shapes into pieces. For example, full STL meshes usually contain several pieces (meshes). Scission will split that into the smaller pieces, and let designs use any piece. It decides where to cut by traversing the polygons, and assigning each polygon an index, i.e. id for the piece. Then it reassembles by id.
